src/main/java/com/conveyal/r5/transit/TransitLayer.java

```diff
@@ -531,9 +882,33 @@ public void rebuildTransientIndexes () {
             }
         }
 
+        if (!fareLegRules.isEmpty()) rebuildFaresV2TransientIndices();
+
         LOG.info("Done rebuilding transient indices.");
     }
 
+    /**
+     * Rebuild transient indices used in Fares V2 routing
+     */
+    private void rebuildFaresV2TransientIndices () {
+        fareLegRulesForFromStopId = indexFareLegRulesForStops(fareLegRulesForFromAreaId);
+        fareLegRulesForToStopId = indexFareLegRulesForStops(fareLegRulesForToAreaId);
+    }
+
+    /** Build an index for which fare leg rules are applicable for trips at each stop. Used for both from and to stops
+     * by passing in fareLegRulesForFromAreaId or fareLegRulesForToAreaId, respectively.
+     */
+    private TIntObjectMap<RoaringBitmap> indexFareLegRulesForStops(TIntObjectMap<RoaringBitmap> fareLegRulesForFareAreaId) {
+        TIntObjectMap<RoaringBitmap> forStops = new TIntObjectHashMap<>();
+        Map<RoaringBitmap, RoaringBitmap> bitmaps = new HashMap<>();
```

Comment:
Now unused?

Reply:
yes, now unused. I have a commit locally that fixes this but I'm not a collaborator on `analysis-backend` so can't push.

Reply:
fixed.